diff --git a/recipes-asteroid/asteroid-launcher/asteroid-launcher_git.bb b/recipes-asteroid/asteroid-launcher/asteroid-launcher_git.bb
index 43052cd..3ec0da7 100644
--- a/recipes-asteroid/asteroid-launcher/asteroid-launcher_git.bb
+++ b/recipes-asteroid/asteroid-launcher/asteroid-launcher_git.bb
@@ -3,7 +3,7 @@ HOMEPAGE = "https://github.com/AsteroidOS/asteroid-launcher"
 LICENSE = "BSD"
 LIC_FILES_CHKSUM = "file://qml/MainScreen.qml;beginline=1;endline=29;md5=3d250dd089f5d6221d9054029963e332"
 
-SRC_URI = "git://github.com/AsteroidOS/asteroid-launcher.git;protocol=https \
+SRC_URI = "git://github.com/MagneFire/asteroid-launcher.git;protocol=https \
     file://asteroid-launcher.service \
     file://default.conf"
 SRC_URI_append_qemux86 = " file://qemu.conf file://kms-qemu.json"
